feature_engine/encoding/decision_tree.py

```diff
@@ -133,17 +134,18 @@ def __init__(
         cv: int = 3,
         scoring: str = "neg_mean_squared_error",
         param_grid: Optional[dict] = None,
-        regression: bool = True,
+        regression: bool = False,
         random_state: Optional[int] = None,
         variables: Union[None, int, str, List[Union[str, int]]] = None,
         ignore_format: bool = False,
+
     ) -> None:
 
         self.encoding_method = encoding_method
         self.cv = cv
         self.scoring = scoring
-        self.regression = regression
         self.param_grid = param_grid
+        self.regression = regression
```

Comment:
regression was under scoring. Pls check the original file for position and default values.